tests/templates/test_layers.py

```diff
@@ -642,3 +644,173 @@ def circuit(weights):
         expectations = circuit(weights)
         for exp, target_exp in zip(expectations, target):
             assert exp == target_exp
+
+
+class TestParticleConservingU1:
+    """Tests for the ParticleConservingU1 template from the
+    pennylane.templates.layers module."""
+
+    @staticmethod
+    def _wires_gates_u1(wires):
+        """Auxiliary function giving the wires that the elementary gates decomposing
+        ``u1_ex_gate`` act on."""
+
+        exp_wires = []
+
+        exp_wires.append(wires)
+        exp_wires.append(wires)
+
+        exp_wires.append(wires[1])
+        exp_wires.append(wires)
+        exp_wires.append(wires[1])
+        exp_wires.append(wires)
+        exp_wires.append(wires[0])
+
+        exp_wires.append(wires[::-1])
+        exp_wires.append(wires[::-1])
+
+        exp_wires.append(wires)
+        exp_wires.append(wires)
+
+        exp_wires.append(wires[1])
+        exp_wires.append(wires)
+        exp_wires.append(wires[1])
+        exp_wires.append(wires)
+        exp_wires.append(wires[0])
```

Comment:
Not so important, but would it not be cleaner to just define the list as in `exp_wires = [wires, wires, wires[1],...]`?

Reply:
Probably yes, I guess I wanted to avoid a relatively big list in the middle of the unit test. I will look at this.

Reply:
Hey @mariaschuld, notice that here `exp_wires = self._wires_gates_u1(nm_wires[i])` ``nm_wires[i]`` is an array with the indices of the qubits ``n`` and ``m`` that the exchange gate acts on. I can build a list in the loop, as you suggest, but I think this will make the test harder to follow.

Reply:
i wasn't thinking of a loop, but just define a list without the append function. but this is minor...

Reply:
oh! @mariaschuld I had totally missed your suggestion from the very beginning. Sorry about this. I thought you were thinking about removing that helper function.
The list is defined now.